omegaconf/dictconfig.py

```diff
@@ -401,12 +400,20 @@ def __delitem__(self, key: DictKeyType) -> None:
                 ),
             )
 
-        del self.__dict__["_content"][key]
+        try:
+            del self.__dict__["_content"][key]
+        except KeyError:
+            full = self._get_full_key(key=key)
+            if full != key:
+                msg = f"Key not found: '{key}' (path: '{full}')"
+            else:
+                msg = f"Key not found: '{key}'"
+            self._format_and_raise(key=key, value=None, cause=ConfigKeyError(msg))
```

Comment:
The full key will be contained in the exception anyway.
Also, you can retype the exception with tyep_override (I think):
```suggestion
        except KeyError as e:
            msg = f"Key not found: '{key}'"
            self._format_and_raise(key=key, value=None, cause=e, type_override=ConfigKeyError)
```

Reply:
I was able to simplify to
```python
        try:
            del self.__dict__["_content"][key]
        except KeyError:
            msg = "Key not found: '$KEY' (path: '$FULL_KEY')"
            self._format_and_raise(key=key, value=None, cause=ConfigKeyError(msg))
```

I did not have success with setting `cause=e` and `type_override=ConfigKeyError` because `Node._format_and_raise` sets `msg=str(cause)` which in this case would be contain the correct string template (since the original `cause` is of type `KeyError`, not type `ConfigKeyError`).

Reply:
Okay. I still think you can drop the (path: ..) component. it's going to be in the exception anyway. (try to print the resulting exception).

Reply:
Done, in https://github.com/omry/omegaconf/pull/484/commits/6211d9b2899a7197fbc1131b9c81a2ae3ddffb28.